src/custom-element.js

```diff
@@ -934,32 +934,23 @@ function createBaseCustomElementClass(win) {
         this.resources_ = resourcesForDoc(this.ampdoc_);
       }
       if (!this.everAttached) {
+        this.everAttached = true;
```

Comment:
Indeed a maze here. However, from what I can tell, moving `everAttached = true` here (though definitely the right thing to do, imo), creates a risk condition where `resources.upgraded(this)` can be called before `resources.add(this)`. Is that ok?